src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1531,7 +1531,22 @@ function ($a) {
                         ->arrayNode('stop_worker_on_signals')
                             ->defaultValue([])
                             ->info('A list of signals that should stop the worker; defaults to SIGTERM and SIGINT.')
-                            ->integerPrototype()->end()
+                            ->beforeNormalization()
+                                ->always()
+                                ->then(function ($v) {
+                                    return array_map(static function ($element): int {
+                                        $element = \is_array($element) ? $element['signal'] : $element;
+                                        if (\is_string($element)) {
+                                            $signal = \constant('\\'.$element);
```

Comment:
no need for the `'\\'` prefix + we might throw a nicer exception when the const doesn't exist? also, should be check that this is a `SIG*` const?

Reply:
That's a good idea! May we check their existence against `get_defined_constants(true)['pcntl']`?